kernel/src/utils/vec.rs

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+//
+// Copyright (c) 2025 Red Hat, Inc
+//
+// Author: Tyler Fanelli <tfanelli@redhat.com>
+
+extern crate alloc;
+
+use alloc::{collections::TryReserveError, vec::Vec};
+
+/// Allocate a generic Vec of a certain size and copy data from a slice into it. Alternative to the
+/// `.to_vec` method for slices that panics if enough size to allocate the corresponding Vec isn't
+/// available.
+pub fn try_to_vec<T: Copy>(input: &[T]) -> Result<Vec<T>, TryReserveError> {
+    let mut vec = Vec::new();
+    let len = input.len();
+
+    vec.try_reserve_exact(len)?;
+    vec.extend_from_slice(input);
+
+    Ok(vec)
+}
+
+/// Allocate a u8 Vec of a certain size. Alternative to the `vec!` macro that panics if enough bytes
+/// to allocate the corresponding Vec isn't available.
+pub fn u8_vec_sized(size: usize) -> Result<Vec<u8>, TryReserveError> {
+    let mut vec: Vec<u8> = Vec::new();
+
+    vec.try_reserve_exact(size)?;
+    vec.resize(size, 0);
+
+    Ok(vec)
+}
```

Comment:
I would suggest to use generics, something like this:

```suggestion
pub fn vec_sized<T: Default + Clone>(size: usize) -> Result<Vec<T>, TryReserveError> {
    let mut vec: Vec<T> = Vec::new();

    vec.try_reserve_exact(size)?;
    vec.resize(size, T::default());

    Ok(vec)
}
```

I'm not sure about `T::default()` or just pass the init value as parameter to the function, up to you.

Reply:
Yes, much nicer! Fixed.